crates/eth-rpc/src/rpc.rs

```diff
@@ -1,34 +1,49 @@
+use std::collections::HashMap;
+use std::marker::PhantomData;
 use std::sync::Arc;
 
-use jsonrpsee::RpcModule;
+use jsonrpsee::core::Error;
+use jsonrpsee::{Methods, RpcModule};
 use kakarot_rpc_core::client::api::KakarotEthApi;
 use starknet::providers::Provider;
 
 use crate::api::alchemy_api::AlchemyApiServer;
 use crate::api::eth_api::EthApiServer;
-use crate::servers::eth_rpc;
+use crate::servers::alchemy_rpc::AlchemyRpc;
 use crate::servers::eth_rpc::KakarotEthRpc;
-use crate::servers::token_rpc::{self, TokenRpc};
 
-pub struct KakarotRpc<P: Provider + Send + Sync + 'static> {
-    eth_rpc_module: RpcModule<KakarotEthRpc<P>>,
-    token_rpc_module: RpcModule<TokenRpc<P>>,
+/// Represents RPC modules that are supported by reth
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub enum KakarotRpcModule {
+    Eth,
+    Alchemy,
 }
 
-impl<P: Provider + Send + Sync + 'static> KakarotRpc<P> {
+pub struct KakarotRpcModuleBuilder<P: Provider + Send + Sync + 'static> {
+    modules: HashMap<KakarotRpcModule, Methods>,
+    _phantom: PhantomData<P>,
```

Comment:
why do we need phantom data?

Reply:
Without it the compiler will throw error for type parameter `P`

<img width="1193" alt="image" src="https://github.com/kkrt-labs/kakarot-rpc/assets/41180869/f95e74c2-e89b-45c0-b09d-a4f95b05f1ea">